frontend/src/components/Collections/components/Dataset/components/DownloadDataset/components/Content/components/CurlLink/components/CopyButton/index.tsx

```diff
@@ -0,0 +1,95 @@
+import copy from "clipboard-copy";
+import { useEffect, useRef, useState } from "react";
+import { CopyButton as Button } from "./style";
+import { Fade } from "@mui/material";
+import { ANIMATION, ANIMATION_STEP } from "./constants";
+
+interface Props {
+  curl: string;
+  handleAnalytics: () => void;
+}
+
+export default function CopyButton({
+  curl,
+  handleAnalytics,
+}: Props): JSX.Element {
+  const [animationStep, setAnimationStep] = useState<ANIMATION_STEP>(
+    ANIMATION_STEP.IDLE
+  );
+  const timeoutRef = useRef<NodeJS.Timer>();
+  const animation = ANIMATION[animationStep];
+
+  // Copy to clipboard, handle analytics, and initiate the copy animation.
+  const handleCopyClick = () => {
+    if (animation.isCopying) {
+      return; // Copying is in progress.
+    }
+    copy(curl);
+    handleAnalytics();
+    setAnimationStep(incrementAnimationState);
+  };
+
+  // Callback fired after the "exited" or "entered" status is applied.
+  // Executes while copy animation is in progress for "COPY_EXIT", "COPIED_ENTER" and "COPIED_EXIT", "COPY_ENTER".
+  // Increments the animation step.
+  const onUpdateAnimationStep = () => {
+    timeoutRef.current = setTimeout(() => {
```

Comment:
not sure if this is helpful, but we have `react-use` package, so we can use `useTimeoutFn` here if needed!

https://github.com/streamich/react-use/blob/master/docs/useTimeoutFn.md